src/spring-cloud/azext_spring_cloud/_enterprise.py

```diff
@@ -229,27 +233,41 @@ def _build_and_get_result(cmd, client, resource_group, service, name, version, a
     logger.warning("[3/{}] Creating or Updating build '{}'.".format(total_steps, name))
     build_result_id = _queue_build(client, resource_group, service, name, relative_path, target_module)
     logger.warning("[4/{}] Waiting for building docker image to finish. This may take a few minutes.".format(total_steps))
-    _wait_build_finished(cmd, client, build_result_id)
+    _wait_build_finished(cmd, client, service, build_result_id)
     return models.BuildResultUserSourceInfo(version=version, build_result_id=build_result_id)
 
 
-def _wait_build_finished(cmd, client, build_result_id):
+def _wait_build_finished(cmd, client, service, build_result_id):
     resource_id = parse_resource_id(build_result_id)
     resource_group = resource_id['resource_group']
     service = resource_id['name']
     build_service = resource_id['child_name_1']
     build = resource_id['child_name_2']
     build_result_name = resource_id['resource_name']
 
+    # Wait until build pod started
     progress_bar = cmd.cli_ctx.get_progress_controller()
-    result = client.build_service.get_build_result(resource_group, service, build_service, build, build_result_name)
-    progress_bar.add(message=result.properties.status)
     progress_bar.begin()
-    while result.properties.status == "Building" or result.properties.status == "Queuing":
+    result = client.build_service.get_build_result(resource_group, service, build_service, build, build_result_name)
+    while (not result.properties.build_pod_name or not result.properties.build_stages) and (result.properties.status == "Building" or result.properties.status == "Queuing"):
```

Comment:
still needs `and (result.properties.status == "Building" or result.properties.status == "Queuing")`? I feel it can check only pod_name and build stages here?

Reply:
We might still need it because let's say the backend crashed by some reasons, and `status` is `"Failed"` but both `build_pod_name` and `build_stages` are `null`. We don't want to enter the infinity loop here.